src/NUnitTestAdapter/Internal/NavigationDataHelper.cs

```diff
@@ -17,6 +17,20 @@ public void LoadAssembly(string assemblyName)
             targetAssembly = Assembly.LoadFrom(assemblyName);
         }
 
+        public string GetDefiningClassName(string className, string methodName)
+        {
+            var type = targetAssembly.GetType(className);
+
+            if (type != null)
+            {
+                var method = type.GetMethod(methodName, BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
+                if (method != null)
+                    return method.DeclaringType.FullName;
+            }
+
+            return className;
+        }
+
```

Comment:
We need both methods called in this order because you could have an async method defined in a base class.